### DISCUSSION TITLE
Broken "how_tos" links?! Lost valuable resource?

### DISCUSSION THREAD

**Discussion by duraluminium** [Koenkk/zigbee2mqtt#9450] [General]
I was just browsing some of my bookmarks looking for instructions on setting up z2mqtt stuff and noticed all links

`https://www.zigbee2mqtt.io/how_tos/xxx`

specifically [How to sniff zigbee traffic](https://www.zigbee2mqtt.io/how_tos/how_to_sniff_zigbee_traffic.html)

have become broken and lost their content. Previously these pages held very valuable instructions on how to get many things in the ecosystem working. Have these pages been migrated to some other place? Navigating to the link just gives a 404 error and no other help. I hope this treasure trove of information has not gotten lost/deleted.
Can anyone comment on either what happened to those pages or even better fix the broken links/reroute?
Thanks!

**Comment by Koenkk** [comment]
Site has been redesigned, new link https://www.zigbee2mqtt.io/advanced/zigbee/04_sniff_zigbee_traffic.html#with-cc2531

**Reply by duraluminium** [reply]
Ah thanks for the quick reply. It's a pity that the old links don't automatically point to the new address, or at least use the same title so one could just copy-paste old link titles to the new addresses. I guess nothing a bit of searching will fix. Thanks!

**Reply by psi-4ward** [reply]
I thought about keeping redirect pages for a while but:
* it's much work 
* it will pollute the repo and it becomes harder to find the correct page
* we have to exclude the old pages from search-indexer 

A good solution would be a redirect-map but something like that is not supported on github-pages.

**Reply by michal-bartak** [reply]
> we have to exclude the old pages from search-indexer

301 HTTP redirect resolves that.

**Reply by psi-4ward** [reply]
In theory but gh pages does not support redirects

**Comment by jk** [comment]
Not only that: Links from the zigbee2mqtt web UI points into to void and error messages on stdout that also points to URL (in my case, the USB dongle wasn't found anymore as I updated from 1.15 to 1.22) point to the FAQ with a `#help-zigbee2mqtt-fails-to-start` anchor - the new FAQ did miss that specific content.

**Reply by smokey42** [reply]
I've got an example just now. The Web UI links to for example https://www.zigbee2mqtt.io/devices/TS011F_plug.html#tuya-ts011f_plug

The Google results have been updated already, so there's no problem.

Another solution to this problem could be a custom 404 page containing a mapping to the old URIs which is doing a JavaScript redirect. This could alleviate the pain for the Web UI users.

**Reply by psi-4ward** [reply]
Device links should still work

https://www.zigbee2mqtt.io/devices/PECWS01.html

**Reply by psi-4ward** [reply]
Think the devices for tuya got renamed?

https://www.zigbee2mqtt.io/supported-devices/#s=Ts011&v=TuYa

**Reply by smokey42** [reply]
The link went to https://www.zigbee2mqtt.io/devices/TS011F_plug.html but in the database there are only https://www.zigbee2mqtt.io/devices/TS011F_plug_1.html, https://www.zigbee2mqtt.io/devices/TS011F_plug_2.html and https://www.zigbee2mqtt.io/devices/TS011F_plug_3.html

**Reply by smokey42** [reply]
So this may be a different problem.

**Reply by Koenkk** [reply]
That is not due to the website redesign, we had to separate these TS0111F plugs because some support power measurements via reporting and some via polling.

**Comment by bkarlson** [comment]
https://www.zigbee2mqtt.io/guide/configuration/webui.html

this also
